sdk/storage/azure_storage_blob/tests/blob_client.rs

```diff
@@ -576,6 +582,23 @@ async fn test_encoding_edge_cases(ctx: TestContext) -> Result<(), Box<dyn Error>
         let properties = blob_client_new.get_properties(None).await?;
         assert_eq!(17, properties.content_length()?.unwrap());
 
+        // Check Name Equality
+        let mut list_blobs_response = container_client.list_blobs(None)?;
+        let page = list_blobs_response.try_next().await?;
+        let list_blob_segment_response = page.unwrap().into_body()?;
+        let blob_items = list_blob_segment_response.segment.blob_items;
+        assert_eq!(1, blob_items.len());
+        let listed_blob_name = blob_items[0]
+            .name
+            .as_ref()
+            .unwrap()
+            .content
+            .as_ref()
+            .unwrap();
```

Comment:
This feels way too cumbersome to do a simple operation like getting the blob name out...